### DISCUSSION TITLE
xml2gff GFF v4 support?

### DISCUSSION THREAD

**Discussion by kanasc** [xoreos/xoreos#585] [Q&A] [answered]
Hi there!

I wondered if there were still plans to add v4 GFF writing for xml2gff.exe? I love these tools and figured they'd be a fantastic way to do batch edits to GFF files (I'm modding Dragon Age: Origins), and gff2xml worked wonderfully. But after I'd made my edits and went to use xml2gff, I got "ERROR: TODO: Add GFF4 writer support" (which I should have seen from the documentation of course).

Thank you for making this project!

**Comment by DrMcCoy** [comment, marked as answer]
Yes, that's theoretically a thing I do want to add. It's just that I don't really have much time to work on xoreos lately, for various personal reasons.

And I admit, I'm even feeling a bit of a burn-out related to xoreos, mainly because I'm mostly in it alone. I wished for xoreos to be more of a communal project, with many hands working on different parts. This hasn't happened, unfortunately. I mean, there are contributors, and I value every single one of them, but there's no denying the project is going slow. It is what it is, I guess.

So the honest answer regarding adding GFFv4 support to xml2gff I can give is that, yes, it's a planned feature, but it's unlikely that it is coming soon. Realistically, it would have better chances if someone else than me worked on it. If you think you have the know-how and will to do that, you are welcome to to that and I'd seriously appreciate it. I'd be happy to give you some pointers, answer questions, give some ideas on how it'd integrate with the current codebase, etc. No pressure, though, I'm not the boss of you, I can't force you if that's not your cup of tea :P

**Reply by kanasc** [reply]
Thanks so much for your quick reply! (And sorry mine is less so!)

I'm sorry this project hasn't been what you envisaged in terms of contributors; I'm not surprised you're feeling burnt out with such a large project on your shoulders. I appreciate the invitation to contribute, but unfortunately I've also got a lot of personal things going on at the moment that prevent me from being reliable on large projects, let alone gathering the know-how in the first place, since I don't really possess it at the moment (it's been a long time since I've worked in C++ even if I enjoyed it very much and would like to return to it someday). I would love to join you someday but I don't want to get your hopes up since my current situation is going to take a very long time to resolve.

I wish you luck in the meantime and I'll try to see if I can send people your way! I found a different solution for the project I mentioned (fbxcmd plus dao's own mmh processors), but I'll be using xoreos for a GFF3 project.
